public_html/wp-content/plugins/wordcamp-payments/includes/sponsor-invoice.php

```diff
@@ -419,8 +419,89 @@ function save_invoice( $post_id, $post ) {
 			update_post_meta( $post_id, $meta_key, $value );
 		}
 	}
+
+	$should_schedule_invoice_reminder = apply_filters( 'should_schedule_invoice_reminder', true );
+	if ( $should_schedule_invoice_reminder && 'wcbsi_approved' === $post->post_status ) {
+		// Schedule organizer reminder if its not already scheduled.
+		$args = array( $post_id );
+		if ( ! wp_next_scheduled( 'send_invoice_pending_reminder', $args ) && ! get_post_meta( $post_id, 'last_reminder_details', true ) ) {
+			wp_schedule_single_event( time() + DAY_IN_SECONDS * 30, 'send_invoice_pending_reminder', array( $post_id ) );
+			update_post_meta( $post_id, 'reminder_schedule_start_time', time() );
+		}
+	}
+}
+
+/**
+ * Send reminder to organizer about the unpaid invoice.
+ *
+ * @param int $invoice_post_id ID of the sponsor invoice post.
+ */
+function send_invoice_pending_reminder_action( $invoice_post_id ) {
+
+	$invoice = get_post( $invoice_post_id );
+
+	$status = $invoice->post_status;
+
+	if ( 'wcbsi_approved' !== $status ) {
+		// Looks like Invoice is not pending anymore. Nothing to do.
+		return;
+	}
+
+	$last_reminder = get_post_meta( $invoice_post_id, 'last_reminder_details', true );
+
+	// We will send reminders after 30, 45, and 60 days.
+	$reminder_schedule = array(
+		1 => 30,
+		2 => 15,
+		3 => 15,
+	);
+
+	$current_reminder_details = array();
+	if ( empty( $last_reminder ) ) {
+		// This is the first reminder that goes out after 30 days.
+		$current_reminder_details['step'] = 1;
+	} else {
+		$current_reminder_details['step'] = $last_reminder['step'] + 1;
+	}
+	$current_reminder_details['sent_at'] = time();
+
+	$edit_link = get_site_url() . "/wp-admin/post.php?post=$invoice_post_id&action=edit";
+
+	$mail = <<<HTML
```

Comment:
What do you think about using something like this instead of heredoc?

```php
$message = str_replace( "\t", '', "
	The invoice for `{$invoice->post_title}` $status_message

	You can view the invoice and its status any time at:

	$invoice_url
	$attachment_message

	If you have any questions, please reply to let us know."
);
```

That's from `WordCamp\Budgets_Dashboard\Sponsor_Invoices\notify_organizer_status_changed()`. I like it because it doesn't break the indentation pattern. Not a big deal, though.

Reply:
:+1: